### PR TITLE
CDH: large buffer in ImageProcess

### PR DESCRIPTION
**Pull request by s026134**
Issue: #172

Added a large buffer to the ImageProcess class for the Onboard Data Handler to improve inter-board data rate by writing to the SD card in blocks of 512 bytes instead of every time we receive data. It is a circular buffer that will collect the logs and then when it hits 512 bytes in the buffer it will write to a file. Any remaining data will be added to the buffer after the write and index variable keeps track of the current size.

### CODE REVIEW

File: flight/core/data_handler.py [hunk 1]
```diff
@@ -570,7 +570,9 @@ def __init__(self, tag_name: str):
         self.current_path = self.create_new_path()
         self.delete_paths = []  # Paths that are flagged for deletion
         self.excluded_paths = []  # Paths that are currently being transmitted
-        self.circular_buffer_size = 20  # Default size of the circular buffer for the files in the directory
+        self.circular_buffer_size = 512  # Default size of the circular buffer for the files in the directory
+        self.img_buf = bytearray(self.circular_buffer_size)  # Pre-allocated static buffer for image process
+        self.img_buf_index = 0  # index to track position in buffer
```
Comment (Ibrassow) [inline flight/core/data_handler.py:576]: use another variable, the circular_buffer_size here is used for the cleaning of the files

File: flight/core/data_handler.py [hunk 2]
```diff
@@ -604,19 +606,39 @@ def create_new_path(self) -> str:
...
+            self.file.flush()
+
+            # Store the over flow data for next time buf is full
+            overflow_len = data_len - space_remaining
+            self.img_buf[:overflow_len] = data[space_remaining:]
+            self.img_buf_index = overflow_len  # set the index to the overflow value
+        else:
+            # Nominal - just adding to the buffer until full
+            self.img_buf[self.img_buf_index : self.img_buf_index + data_len] = data
+            self.img_buf_index += data_len  # increment index for the buffer
```
Comment (Ibrassow) [inline flight/core/data_handler.py:639]: Make all this conditioned on cls.SD_ERROR_FLAG being false, so in the case the SD card fails, we will be able to transmit slowly (and painfully) through the RAM to the ground

File: tests/test_data_handler.py [hunk 3]
```diff
@@ -70,13 +72,37 @@ def test_extract_time_from_filename(input_filename, expected_output):
...
+@pytest.fixture
 def sd_root(tmpdir):
-    sd_root = tmpdir.mkdir("sd")
+    sd_root = tmpdir.mkdir(dh._HOME_PATH)
     return sd_root
 
-def test_scan_sd_card(sd_root):
-    DH.sd_path = str(sd_root)"""
+
+def test_image_log(sd_root):
```
Comment (Ibrassow) [inline tests/test_data_handler.py:82]: Need also to test failing and edge cases

**Review by Ibrassow — CHANGES_REQUESTED** [review]

**Comment by Ibrassow** [comment]
Can you change the target branch to main?

**Review by Ibrassow — APPROVED** [review]
